cfe_internal/core/watchdog/templates/watchdog.mustache

```diff
@@ -1,8 +1,152 @@
 #!/usr/bin/sh
 
-# If cf-execd is present and executable, and there is no cf-execd process running, then launch it.
+set -e
 
-[ -x {{{vars.sys.cf_execd}}} ] && if ! ( ps -ef | grep -q {{{vars.sys.bindir}}}/[c]f-execd ); then
-  {{{vars.sys.cf_execd}}}
+# Watchdog events are logged here.
+LOGFILE="/tmp/watchdog.log"
+touch ${LOGFILE}
+TMPDIR="/tmp"
+CFENGINE_WORKDIR="/tmp/var/cfengine"
+CFENGINE_WORKDIR_COLLECTION=""
+OUTPUTS_DIR="${CFENGINE_WORKDIR}/outputs"
+ARCHIVE_DIR="/tmp/cfengine-watchdog-archives"
+mkdir -p "${ARCHIVE_DIR}"
+PATHOLOGY_COUNT=0
+PATHOLOGY_THRESHOLD=1
+
+# Collection Dir
+# - We create a directory in order to collect artifacts about our observations
+# - If there are enough pathology indicators found to warrant a report, the
+#   directory will be packaged for sending
+# Portable mktemp: https://stackoverflow.com/questions/10224921/how-to-create-a-temporary-file-with-portable-shell-in-a-secure-way#comment86787877_10235393
+# Adjusted, known to work on aix 7.1.0.0
+if [ -r "/dev/urandom" ]; then
+    RNDM="/dev/urandom"
+else
+    RNDM="/dev/random"
+fi
+length=7; safetemp=$(od -An -N${length} -tx1 ${RNDM} | tr -d ' \t').cfengine-watchdog
+COLLECTION_DIR="${TMPDIR}/${safetemp}"
+mkdir -p "${COLLECTION_DIR}"
+COLLECTION_REPORT="${COLLECTION_DIR}/README.org"
+echo "#+Title: CFEngine Watchdog Engineering Summary [$(date '+%Y-%m-%d %a')]" >> ${COLLECTION_REPORT}
+echo "* Pathologies" >> ${COLLECTION_REPORT}
+
+
+# Pathology #1: cf-execd is not running.
+# While not strictly a pathology, it is non-standard to run cf-agent without cf-execd.
+if ( ps -ef | grep -q \/var\/cfengine\/bin\/[c]f-execd ); then
+    echo "$(date) Found cf-execd not running" >> ${LOGFILE}
```

Comment:
what about using an alternate workdir? Maybe set a value at top that can be adjusted and then use it everywhere else?